### ISSUE TITLE
Unexpected behaviour when `@{dict}` keys have period characters (`"."`)

### ISSUE THREAD

**Issue by tanghyd** [streamsync-cloud/streamsync#24]
I'm really liking the app so far! I just wanted to log this in case it would be helpful - feel free to close if its not a priority.

## Bug Overview

The string keys of dictionaries used in radio input components appear to have unexpected behaviour when they contain periods (`"."`), forward slashes (`"/"`), or similar path-like values.

EDIT: I have a hunch that there may be some strange logic with `.split(".")` calls. This bug might only be due to `"."` characters, rather than forward slashes. See: 
- https://github.com/ramedina86/streamsync/blob/master/src/streamsync/core.py#L609
- https://github.com/ramedina86/streamsync/blob/master/src/streamsync/core.py#L621

### Short Example

This is the initial state of the dashboard UI when I initialize the application:

- Refresh button: triggers the standard `update` function on click.
- Radio items: maps to the `state["files"]` variable with `@{files}`, triggers `_select_file_handler` on click.
- Top header "Selected Files": "None" if nothing is selected, otherwise it is the _value_ of the selected item from the `state["files"]` dictionary.

![app_start](https://github.com/ramedina86/streamsync/assets/34246079/fb87ab85-360e-4ff7-8485-6f7fcc8537db)

If I have a period or forward slash character in my radio input dictionary, then after clicking on a radio item (which I've mapped to a handler) or hitting the "refresh" button (which I have mapped to the standard `update` function), it silently breaks the list of radio inputs and shows something else:

![select_item_with_slash_and_period](https://github.com/ramedina86/streamsync/assets/34246079/dc40e516-b05b-4a9a-b726-dae20c216614)

By "radio input dictionary", I mean to say the input of the radio input is `@{files}` and I have defined a key called `files` in the `state: StreamsyncState` object as follows:

```python
state["files"] = {"../data/hello_world.txt": "../data/hello_world.txt"}
```

Note that the issue above is only related to the dictionary keys - the dictionary values can be any arbitrary string without issue.

All source code to produce this error is attached below.

### Ideal Outcome

I would like to use strings with path characters (i.e. periods, forward slashes) in my radio item keys so that I can have users select a particular file they might like to load. I can work around this behaviour, but I think that at the very least I think this behaviour should be documented, as I can't see anything in the documentation stating that this causes issues.

### Reproducible Example

I have included a minimal reproducible example as a streamsync folder that can be run and checked against. Please see the README in this .zip file for more details as I have broken down some of the error cases as best I can.

Once a virtual environment is activated, you should be able to reproduce the issue described in the README file when running either `streamsync run app` or `streamsync edit app`.

#### main.py file

The entry point `main.py` file is printed below:

```python
"""Entry point script for the streamsync dashboard application."""

from pathlib import Path
from typing import Iterable

import streamsync as ss
from streamsync.core import StreamsyncSession, StreamsyncState

DATA_DIR = "../data"


def update(state: StreamsyncState, session: StreamsyncSession):
    """Updates the application state."""
    _update_files(state)


def _update_files(state: StreamsyncState):
    """Updates the list of discovered data files in the application state."""
    paths: Iterable[Path] = Path(state["data_dir"]).glob("*.txt")
    state["files"]: dict[str, str] = {str(path): path.name for path in paths}


def file_select_handler(state: StreamsyncState, payload: str):
    """Handler function when a file select component is interacted with."""
    state["selected_file"] = payload


# Initialise the dashboard application state
initial_state = ss.init_state(
    {
        "my_app": {"title": "Streamsync Dashboard"},
        "data_dir": DATA_DIR,
        "files": {},
        "selected_file": "None",
    }
)

update(initial_state, None)

```

#### Compressed Project as a Zip File

The full project is attached here: [bug_report.zip](https://github.com/ramedina86/streamsync/files/11469055/bug_report.zip)

#### Project File Contents

A screenshot of the folder contents is shown below:

![image](https://github.com/ramedina86/streamsync/assets/34246079/b54668fb-9045-4722-b888-ada600320159)

**Comment by ramedina86** [comment]
Thank you so much for taking the time to report this in such a thorough way. It's definitely not getting closed. This is likely a bug, I'll have to more digging to confirm and fix it for the next release. If it's not a bug, as you suggested, I'll make sure to improve the documentation.

**Comment by ramedina86** [comment]
Fixed in 0.1.7, improved test cases to avoid regression.
